### PR TITLE
refactor: use gcloud to deploy

### PR DESCRIPTION
**Pull request by averikitsch**

### CODE REVIEW

File: src/deploy-cloudrun.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        region,
+      ];
+    }
+    if (!metadata) {
+      // Set optional flags from inputs
+      if (envVars) cmd.push('--update-env-vars', envVars);
+    }
+
+    // Install gcloud if not already installed.
+    if (!setupGcloud.isInstalled()) {
```
Comment (bharathkkb) [inline src/deploy-cloudrun.ts:82]: Should give users an option to pin gcloud version for idempotent runs?

File: src/deploy-cloudrun.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        stdout,
+      },
+      silent: true,
+    };
+
+    // Run gcloud cmd.
+    try {
+      await exec.exec(toolCommand, cmd, options);
+      // Set url as output.
+      setUrlOutput(errOutput);
```
Comment (bharathkkb) [inline src/deploy-cloudrun.ts:133]: Does gcloud output url to stderr?

File: src/deploy-cloudrun.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      } else {
+        throw new Error(error);
+      }
+    }
+  } catch (error) {
+    core.setFailed(error.message);
+  }
+}
+
+function setUrlOutput(output: string): void {
```
Comment (bharathkkb) [inline src/deploy-cloudrun.ts:146]: Can we add some tests for this since it's a regex match?

**Review by bharathkkb — COMMENTED** [review]
Overall LGTM
Should we first publish the sdk, which will reduce the duplication here or should we do that in a follow up?

File: src/deploy-cloudrun.ts (hunk 2 above)
Comment (averikitsch) [reply src/deploy-cloudrun.ts:133]: Yeah a little odd - integration tests should show this working correctly

File: src/deploy-cloudrun.ts (hunk 3 above)
Comment (averikitsch) [reply src/deploy-cloudrun.ts:146]: done

**Comment by averikitsch** [comment]
> Overall LGTM
> Should we first publish the sdk, which will reduce the duplication here or should we do that in a follow up?

I was going to do that as a follow up

File: src/deploy-cloudrun.ts (hunk 1 above)
Comment (averikitsch) [reply src/deploy-cloudrun.ts:82]: sg

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**